### PR TITLE
[JENKINS-68419] Ignore quality gate when threshold is zero.

### PR DESCRIPTION
**Pull request by miked-rr**
Add test to check success when quality gate threshold set to zero, update QualityGateEvaluator to not run evaluation when threshold is zero.

### CODE REVIEW

File: plugin/src/test/java/io/jenkins/plugins/analysis/core/util/QualityGateEvaluatorTest.java [hunk 1]
```diff
@@ -133,6 +133,21 @@ void shouldFailIfSizeIsEqual() {
...
+    void shouldIgnoreThresholdZero() {
+        Logger logger = new Logger();
+
+        IssuesStatisticsBuilder builder = new IssuesStatisticsBuilder();
+
+        QualityGateEvaluator qualityGate = new QualityGateEvaluator();
+
+        qualityGate.add(0, QualityGateType.TOTAL, QualityGateResult.UNSTABLE);
+        assertThat(qualityGate.evaluate(builder.setTotalNormalSize(1).build(), logger)).isEqualTo(QualityGateStatus.PASSED);
+
```
Comment (uhafner) [inline plugin/src/test/java/io/jenkins/plugins/analysis/core/util/QualityGateEvaluatorTest.java:146]: Can you add a check that the log is empty?

**Review by uhafner — CHANGES_REQUESTED** [review]
Thanks for your pull request, just a small improvement to the test would make sense.

**Review by uhafner — APPROVED** [review]
I hope you don't mind, in order to get this PR merged I added the missing assertion on my own 😄

**Comment by miked-rr** [comment]
Of course I don't mind! Thanks!
